src/components/PortfolioBalancesProvider/PrefetchBalancesWrapper.tsx

```diff
@@ -0,0 +1,15 @@
+import { useCachedPortfolioBalances } from 'components/PortfolioBalancesProvider'
+import useHoverProps from 'hooks/useHoverProps'
+import { PropsWithChildren } from 'react'
+
+/* Prefetches & caches portfolio balances when the wrapped component is hovered */
+export default function PrefetchBalancesWrapper({ children, className }: PropsWithChildren<{ className?: string }>) {
```

Comment:
PrefetchBalanceWrapper now just has a single responsbility: fetching on hover. `shouldFetchOnAccountUpdate` was removed, with that functionality being replaced w/ other callers of the `useCachedPortfolioBalances` hook passing the `freshBalancesRequired` flag when necessary